src/DotNetty.Common/Concurrency/IEventExecutor.cs

```diff
@@ -46,5 +46,9 @@ public interface IEventExecutor
         Task ShutdownGracefullyAsync();
 
         Task ShutdownGracefullyAsync(TimeSpan quietPeriod, TimeSpan timeout);
+
+        void Schedule(Action<object, object> action, object state1, object state2, TimeSpan delay);
+
+        void Schedule(Action<object, object> action, object state1, object state2, TimeSpan delay, CancellationToken cancellationToken);
```

Comment:
What's the purpose of having two different state objects in the `Schedule` signature here? Does one have a distinct difference than the other? Or is it totally arbitrary and specific to the `Action` ?

Not challenging the current design, just want to know what's the distinction here :p